src/bitmap/container.rs

```diff
@@ -94,6 +94,37 @@ impl Container {
         result
     }
 
+    pub fn remove_first(&mut self, n: usize) {
+        match &self.store {
+            Store::Bitmap(bits) => {
+                if bits.len() - n as u64 <= ARRAY_LIMIT {
+                    let mut replace_array = Vec::with_capacity(bits.len() as usize - n);
+                    replace_array.extend(bits.clone().into_iter().skip(n));
+                    self.store = Store::Array(store::ArrayStore::from_vec_unchecked(replace_array));
+                } else {
+                    self.store.remove_first(n)
+                }
+            }
+            Store::Array(_) => self.store.remove_first(n),
+        };
+    }
+
+    pub fn remove_last(&mut self, n: usize) {
+        match &self.store {
+            Store::Bitmap(bits) => {
+                if bits.len() - n as u64 <= ARRAY_LIMIT {
+                    let mut replace_array = Vec::with_capacity(bits.len() as usize - n);
+                    replace_array.extend(bits.clone().into_iter());
+                    replace_array.truncate(replace_array.len() - n);
```

Comment:
This should use `replace_array.extend(bits.iter().take(bits.len() as usize - n));`

To avoid cloning bits, and we don't need to over-fill first, then truncate.

Reply:
fixed
https://github.com/RoaringBitmap/roaring-rs/pull/253/commits/d0e7322a07bf3d68369cd6bcf5a5ace62f7711b1